### PR TITLE
Feat: Timesheet time and Half Day leave indicator

### PR DESCRIPTION
**Pull request by ananyo141**
> Please provide enough information so that others can review your pull request:

This PR is to address these issues:
- https://github.com/rtCamp/ERP/issues/1961
- https://github.com/rtCamp/frappe-slack-connector/issues/63
- https://github.com/rtCamp/frappe-slack-connector/issues/64

> Explain the **details** for making this change. What existing problem does the pull request solve?

It fixes 3 problems
- Shows the half day indicator to the reporting manager
- Sends the timesheet notification on given time (not at default 12AM)
- Added the fixture that automatically fetches the Leave Approver Name from `leave_approver` field automatically.

> QA List

- [ ] Check half day indicator is working for both full day and half day leaves
- [ ] Check timesheet notification is sent at the given time
- [ ] Check leave approver name is being properly set

> Screenshots/GIFs

Screenshot 2024-11-06 at 8 48 10 PM

### CODE REVIEW

File: frappe_slack_connector/hooks.py [hunk 1]
```diff
@@ -152,8 +152,10 @@
 # ---------------
 
 scheduler_events = {
-    "all": ["frappe_slack_connector.tasks.attendance_summary.attendance_channel"],
-    "daily": ["frappe_slack_connector.tasks.send_daily_reminder.send_reminder"],
+    "all": [
+        "frappe_slack_connector.tasks.attendance_summary.attendance_channel",
+        "frappe_slack_connector.tasks.send_daily_reminder.send_reminder",
```
Comment (zeel-codder) [inline frappe_slack_connector/hooks.py:157]: Why we do this 🤔 ?

**Review by zeel-codder — COMMENTED** [review]
We can also show half name if it is half day leave than 🤔 ?

**Comment by ananyo141** [comment]
As per discussion with @KanchanChauhan yesterday, we only need to show if half day. @zeel-codder

File: frappe_slack_connector/hooks.py (hunk 1 above)
Comment (ananyo141) [reply frappe_slack_connector/hooks.py:157]: Daily is running everyday at 12AM, so as we implemented the scheduler to run at `all` like in attendance_channel, to have control over the time notification is sent

Comment (ananyo141) [reply frappe_slack_connector/hooks.py:157]: As per discussion, we can change it to hourly too

**Comment by ananyo141** [comment]
There is a catch; although not really essential, but notification can hit slack rate limit if there are a lot of users to be sent notification to.

**Review by zeel-codder — APPROVED** [review]

**Comment by niraj2477** [comment]
What is the limit for slack?

**Comment by ananyo141** [comment]
> What is the rate limit for slack?
@niraj2477 As per testing, it is 150+. But there would be exponential backoff (less once rate limit is hit)

**Comment by ananyo141** [comment]
One solution we can do is intentionally throttle the API requests, to 1 message per second [as per Slack's recommendation](https://api.slack.com/apis/rate-limits#:~:text=However%2C%20we%20do%20recommend%20you%20design%20your%20apps%20with%20a%20limit%20of%201%20request%20per%20second%20for%20any%20given%20API%20call%2C%20knowing%20that%20we%27ll%20allow%20it%20to%20go%20over%20this%20limit%20as%20long%20as%20this%20is%20only%20a%20temporary%20burst.).

It would require approx <3.5 mins to execute the request then (maximum 200 notifications to 200 employees)
@zeel-codder @niraj2477

**Pull request merged**